src/systems/optical_tactile_plugin/OpticalTactilePlugin.cc

```diff
@@ -720,208 +636,123 @@ void OpticalTactilePluginPrivate::DepthCameraCallback(
 }
 
 //////////////////////////////////////////////////
-void OpticalTactilePluginPrivate::UnpackPointCloudMsg(
-                        const ignition::msgs::PointCloudPacked &_msg)
+ignition::math::Vector3f OpticalTactilePluginPrivate::MapPointCloudData(
+  const u_int32_t &_i, const uint32_t &_j)
 {
-  // 3D vector to store XYZ points
-  this->imageXYZ = std::vector<std::vector<std::vector<float>>>
-      (_msg.width(),
-      std::vector<std::vector<float>>(_msg.height(),
-      std::vector<float>(3)));
+  // Initialize return variable
+  ignition::math::Vector3f measuredPoint;
+
+  // Save original position of the pointer
+  char *initialMsgBuffer = this->msgBuffer;
+
+  // Number of bytes from the beginning of the pointer (image coordinates at
+  // 0,0) to the desired (i,j) position
+  uint32_t msgBufferIndex =
+    _j*this->cameraMsg.row_step() + _i*this->cameraMsg.point_step();
+
+  this->msgBuffer += msgBufferIndex;
+  int fieldIndex = 0;
+
+  // X coordinate
+  measuredPoint.X() = *reinterpret_cast<float *>(
```

Comment:
Yeah we should check that.
I understand that you suggest checking that the first time we receive a message from the depth camera? Or is it every time? Probably the former is enough.

Reply:
Yeah just once should be sufficient, to save time as much as possible. Presumably the user would not have two nodes both publishing depth images to the same topic - they can, but that is not usually recommended as it would mess up the topic, timing, etc.

Reply:
e216bb2